### PR TITLE
feat/ Fetch user posts api

### LINKED ISSUE #116: Fetch My posts api (active and draft posts)
Fetch My posts api (active and draft posts)

Relates #72

### PR DESCRIPTION
**Pull request by avinash360**
Closes #116 
## Types of changes
- [x] Add tags to auto close the corresponding issue.
- [x] I followed branch and commit naming convention.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] I have read the **CONTRIBUTING** document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: server/controllers/posts.js [hunk 1]
```diff
@@ -5,6 +5,26 @@ import {
 } from '../validations/posts';
 import joiOptions from '../validations/joiOptions';
 
+const filterPostsQuery = async function filterPostsQuery(type, userId) {
```
Comment (nimish-gupta) [inline server/controllers/posts.js]: Why not directly arrow functions?

File: server/controllers/posts.js [hunk 2]
```diff
@@ -5,6 +5,26 @@ import {
...
+  let query = {};
+  if (type === 'user-published') {
+    query = {
+      author: userId,
+      public: true,
+      published: true,
```
Comment (nimish-gupta) [inline server/controllers/posts.js]: Why not add condition for private posts?

File: server/controllers/posts.js (hunk 1 above)
Comment (avinash360) [reply server/controllers/posts.js]: done

File: server/controllers/posts.js (hunk 2 above)
Comment (avinash360) [reply server/controllers/posts.js]: done

**Pull request merged**